File: src/frontend/test/src/domain/entities/Remise.ts
import type { IReservation } from '@/domain/entities/Reservation';

export interface IRemise {
    id: number;
    nomRemise: string;
    codeRemise: string;
    pourcentageRemise: number;
    reservations?: IReservation[];
}

export class Remise implements IRemise {
    constructor(
        public id: number,
        public nomRemise: string,
        public codeRemise: string,
        public pourcentageRemise: number,
        public reservations?: IReservation[]
    ) {}
}

File: src/backend/NHP_Domain/Entities/Remise.cs
namespace NHP_Domain.Entities;

public class Remise
{
    public long Id { get; set; }
    public String NomRemise { get; set; } = String.Empty;
    public String CodeRemise { get; set; } = String.Empty;
    public int PourcentageRemise { get; set; } = 1;
    
    //OneToMany
    public virtual List<Reservation>? Reservations { get; set; } = new();

    public override string ToString()
    {
        return $"nomRemise: {NomRemise} +" +
               $" pourcentage: {PourcentageRemise} +" +
               $" codeRemise: {CodeRemise}";
    }
}